pulp_container/app/models.py

```diff
@@ -164,17 +171,41 @@ def init_manifest_list_nature(self):
                 self.is_flatpak = True
                 return True
 
-        return False
+        return updated_type
 
     def init_manifest_nature(self):
-        if self.is_bootable_image():
-            self.is_bootable = True
-            return True
-        elif self.is_flatpak_image():
-            self.is_flatpak = True
-            return True
-        else:
-            return False
+        known_types = self.known_types()
+        for manifest_type in known_types.values():
+            func = manifest_type["check_function"]
+            if func():
+                self.type = manifest_type["type"]
+
+                # set custom attributes (is_flatpak, is_bootable) to keep compatibility
+                if manifest_type.get("custom", None):
+                    custom_key_name = list(manifest_type["custom"])[0]
+                    custom_key_value = manifest_type["custom"][custom_key_name]
+                    setattr(self, custom_key_name, custom_key_value)
+
+                return True
+
+        return False
```

Comment:
This is not possible because we cannot determine the type of the image from `media_type`. For some of the natures, we still need to look into the config blob (e.g., for helm charts). 🧷